packages/server/dataloader/RootDataLoader.ts

```diff
@@ -41,79 +39,81 @@ const loaderMakers = {
   ...azureDevOpsLoaders
 } as const
 
-type LoaderMakers = typeof loaderMakers
-export type Loaders = keyof LoaderMakers
+export type Loaders = keyof typeof loaderMakers
 
-type PrimaryLoaderMakers = typeof rethinkPrimaryKeyLoaderMakers
-type PrimaryLoaders = keyof PrimaryLoaderMakers
-type Unprimary<T> = T extends RethinkPrimaryKeyLoaderMaker<infer U> ? DBType[U] : never
-type TypeFromPrimary<T extends PrimaryLoaders> = Unprimary<PrimaryLoaderMakers[T]>
+export type AllPrimaryLoaders =
+  | keyof typeof primaryKeyLoaderMakers
+  | keyof typeof rethinkPrimaryKeyLoaderMakers
+export type RegisterDependsOn = (primaryLoaders: AllPrimaryLoaders | AllPrimaryLoaders[]) => void
 
-type ForeignLoaderMakers = typeof rethinkForeignKeyLoaderMakers
-type ForeignLoaders = keyof ForeignLoaderMakers
-type Unforeign<T> = T extends RethinkForeignKeyLoaderMaker<infer U> ? U : never
-type TypeFromForeign<T extends ForeignLoaders> = TypeFromPrimary<Unforeign<ForeignLoaderMakers[T]>>
-
-/**
- * When adding a new loaders file like {@link atlassianLoaders} or {@link githubLoaders}
- * this type has to include a typeof of newly added loaders
- */
-type CustomLoaderMakers = typeof customLoaderMakers &
-  typeof atlassianLoaders &
-  typeof jiraServerLoaders &
-  typeof pollLoaders &
-  typeof integrationAuthLoaders &
-  typeof primaryKeyLoaderMakers &
-  typeof foreignKeyLoaderMakers &
-  typeof azureDevOpsLoaders &
-  typeof gitlabLoaders &
-  typeof gcalLoaders
-type CustomLoaders = keyof CustomLoaderMakers
-type Uncustom<T> = T extends (parent: RootDataLoader) => infer U ? U : never
-type TypeFromCustom<T extends CustomLoaders> = Uncustom<CustomLoaderMakers[T]>
-
-// Use this if you don't need the dataloader to be shareable
-export interface DataLoaderInstance {
-  get<LoaderName extends Loaders>(loaderName: LoaderName): TypedDataLoader<LoaderName>
+// The RethinkDB logic is a leaky abstraction! It will be gone soon & this will be generic enough to put in its own package
+interface GenericDataLoader<TLoaders, TPrimaryLoaderNames> {
```

Comment:
this is basically `TypedDataLoader`, but the old version took 350ms to load types, this takes ~150ms